src/state/slices/stakingDataSlice/selectors.ts

```diff
@@ -169,14 +221,122 @@ export const selectRewardsByAccountSpecifier = createDeepEqualOutputSelector(
   }
 )
 
+export const selectAllRewardsByDenom = createDeepEqualOutputSelector(
+  selectStakingDataByAccountSpecifier,
+  selectDenom,
+  (stakingData, denom): Record<string, chainAdapters.cosmos.Reward[]> => {
+    if (!stakingData || !stakingData.rewards) return {}
+
+    const rewards = stakingData.rewards.reduce(
+      (acc: Record<string, chainAdapters.cosmos.Reward[]>, current: ValidatorReward) => {
+        if (!acc[current.validator.address]) {
+          acc[current.validator.address] = []
+        }
+
+        acc[current.validator.address].push(
+          ...current.rewards.filter(x => ASSET_ID_TO_DENOM[x.assetId] === denom)
```

Comment:
[gomesalexandre](https://github.com/gomesalexandre) [yesterday](https://github.com/shapeshift/web/pull/1356/files#r840058196)
ditto spread

Reply:
I dont mind the spread here. Its a nice way for peace of mind to be sure something i a copy without needing to be 100% sure about how the other functions work

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/filter

> The filter() method creates a new array with all elements that pass the test implemented by the provided function.